api/src/wfl/module/covid.clj

```diff
@@ -25,8 +18,65 @@
 (defn create-covid-workload!
   [tx {:keys [source sink executor] :as request}])
 
-(defn start-covid-workload!
-  [])
+;; TODO: move this to the queue-based skeleton later
+(defn ^:private snapshot-created?
+  "Check if `snapshot-job-id` has done in TDR,
+   return the resulting snapshot id if so."
+  [snapshot-job-id]
+  (some->> snapshot-job-id
+           (datarepo/job-done?)
+           :id))
+
+(defn ^:private go-create-snapshot!
+  "Create snapshot in TDR from `dataset`, `table`
+   and `row-ids` and write job info as well as rows
+   into `source-details-name` table."
+  [tx dataset table source-details-name [start end] row-ids]
+  (let [columns     (-> (datarepo/all-columns dataset table)
+                        (->> (map :name) set)
+                        (conj "datarepo_row_id"))
```

Comment:
We could cache this I think in the source table to avoid making extra calls - what do you think?